hydromt/models/components/grid.py

```diff
@@ -13,370 +10,152 @@
 from pyproj import CRS
 from shapely.geometry import box
 
-from hydromt import workflows
-from hydromt.gis import raster, utils
+from hydromt._typing.error import NoDataStrategy, _exec_nodata_strat
+from hydromt._typing.type_def import DeferedFileClose
+from hydromt.gis import raster
+from hydromt.gis import utils as gis_utils
+from hydromt.io.readers import read_nc
+from hydromt.io.writers import write_nc
 from hydromt.models.api import Model
+from hydromt.models.components.base import ModelComponent
+from hydromt.models.components.region import _parse_region
+from hydromt.workflows.basin_mask import get_basin_geometry
+from hydromt.workflows.grid import (
+    grid_from_constant,
+    grid_from_geodataframe,
+    grid_from_raster_reclass,
+    grid_from_rasterdataset,
+    rotated_grid,
+)
 
-__all__ = ["GridModel"]
-logger = logging.getLogger(__name__)
+__all__ = ["GridComponent"]
 
 
-class GridMixin(object):
-    # placeholders
-    # xr.Dataset representation of all static parameter maps at the same resolution and
-    # bounds - renamed from staticmaps
-    _API = {"grid": xr.Dataset}
+class GridComponent(ModelComponent):
+    """ModelComponent class for grid components.
 
-    def __init__(self, *args, **kwargs) -> None:
-        super().__init__(*args, **kwargs)
-        self._grid = None  # xr.Dataset()
+    This class is used for setting, creating, writing, and reading grids for a
+    HydroMT model. The grid data in the data property of this class is of the
+    hydromt.gis.raster.RasterDataset type.
```

Comment:
Nice additionnal explanation! We do need that :) a few suggestions to improve:
```suggestion
    This class is used for setting, creating, writing, and reading regular grid data for a
    HydroMT model. The grid component data stored in the ``data`` property of this class is of the
    hydromt.gis.raster.RasterDataset type which is an extension of xarray.Dataset for regular grid.
```